elyra/pipeline/kfp/component_parser_kfp.py

```diff
@@ -126,25 +150,54 @@ def _read_component_yaml(self, registry_entry: SimpleNamespace) -> Optional[Dict
+
+        :param parameter_name: the name of the parameter that will be checked
+        :param component_body: the component YAML contents
         """
-        Change the parameter ref according if it is a KFP path parameter (as opposed to a value parameter)
+        # Get component_body['implementation']['container'] sub-dictionary if it exists
+        component_impl = component_body.get('implementation', {}).get('container', {})
+
+        # Get list of component commands/arguments
+        commands_and_args = component_impl.get("command", []) + component_impl.get("args", [])
+
+        # Loop through dictionary-types only; parameter-based fields will
+        # always be of the format {'inputPath': 'parameter name'}
+        for param_dict in [c for c in commands_and_args if isinstance(c, dict)]:
+            # Check the (single-element) list of values for a
+            # match on the full parameter name given
+            if parameter_name in list(param_dict.values()):
+                # Check whether the first (and only) key contains the
+                # phrase "Path", e.g. inputPath or outputPath
+                if "Path" in list(param_dict.keys())[0]:
+                    return True
+                # Otherwise, assume inputValue for this parameter name
+                # and do not proceed to check other parameter dicts
+                break
+
+        return False
+
+    def determine_type_information(self, parsed_type: str) -> SimpleNamespace:
+        """
+        Takes the type information of a component parameter as parsed from the component
+        specification and returns a new type that is one of several standard options.
+
         """
-        adjusted_type = parameter_type
-        if "implementation" in component_body and "container" in component_body['implementation']:
-            if "command" in component_body['implementation']['container']:
-                for command in component_body['implementation']['container']['command']:
-                    if isinstance(command, dict) and list(command.values())[0] == io_object_name and \
-                            list(command.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
-            if "args" in component_body['implementation']['container']:
-                for arg in component_body['implementation']['container']['args']:
-                    if isinstance(arg, dict) and list(arg.values())[0] == io_object_name and \
-                            list(arg.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
-
-        return adjusted_type
+        data_type_info = super().determine_type_information(parsed_type)
+        if data_type_info.undetermined:
+            if 'inputpath' in data_type_info.parsed_data:
+                data_type_info.data_type = 'inputpath'
+                data_type_info.control_id = "NestedEnumControl"
+                data_type_info.undetermined = False
```

Comment:
Can we add a line here that sets the default for `inputpath` to be undefined? 
```python
data_type_info.default_value = None
```

Reply:
I would agree.  In fact, it "feels" more appropriate to default `default_value` to `None` from the start since its data type is hinted as `Any` and, currently, uses the empty string as the default value - which is only compatible with strings.

Would using `None` for `'inputvalue'` and `'outputpath'` defaults also be appropriate?

I'm thinking the correct default, across all the types, is always `None` and we explicitly set the default to the empty string only where that's appropriate.

Reply:
sure thing

Reply:
@kevin-bates i would say no, but I’m not 100% sure. It’s dependent on the type of form control it’s assigned. Some controls can’t handle an undefined default.

Reply:
In this case it’s always assigned the NestedEnumControl which requires undefined as the default

Reply:
I added @bourdakos1 suggested change (and it solved our issue in https://github.com/elyra-ai/pipeline-editor/pull/161 ). @akchinSTC if you want to change it to a more generic fix feel free